src/Microsoft.Health.SqlServer/Features/Client/SqlConnectionWrapper.cs

```diff
@@ -78,6 +87,34 @@ internal async Task InitializeAsync(string initialCatalog = null, CancellationTo
         }
     }
 
+    /// <summary>
+    /// Open sql connection which is retried up to 3 times if <see cref="SqlException"/> is caught with state 35.
+    /// <remarks>
+    /// Retries have been added to cater to the following exception:
+    /// "A connection was successfully established with the server, but then an error occurred during the login process.
+    /// (provider: TCP Provider, error: 35 - An internal exception was caught)"
```

Comment:
Are we sure this is something we want to retry? The SQL Client library already retries a number of different errors? Do we know why this isn't retried? Is this a "retryable" exception? For example, I see an old issue in the issues page for this problem when moving from .NET 5 -> 6: https://github.com/dotnet/SqlClient/issues/1402. Unsure if it's related, but I am reluctant to add new retries if we aren't quite sure why they're happening.